.gradient/prepare-datasets.sh

```diff
@@ -33,24 +11,17 @@ then
 fi
 
 echo "Starting preparation of datasets"
-# symlink exe_cache files
-exe_cache_source_dir="${PUBLIC_DATASET_DIR}/poplar-executables-pytorch-3-1-1"
-symlink-public-resources "${exe_cache_source_dir}" $POPLAR_EXECUTABLE_CACHE_DIR
-# Symlink squad
-symlink-public-resources "${PUBLIC_DATASET_DIR}/squad" "${HF_DATASETS_CACHE}/squad"
-# Symlink OGB Wiki dataset and checkpoint
-symlink-public-resources "${PUBLIC_DATASET_DIR}/ogbl_wikikg2_custom" "${DATASET_DIR}/ogbl_wikikg2_custom"
-
-# symlink local dataset used by vit-model-training notebook
-symlink-public-resources "${PUBLIC_DATASET_DIR}/chest-xray-nihcc-3" "${DATASET_DIR}/chest-xray-nihcc-3"
+cd "$(dirname "$0")"
+python -u ./symlink_datasets_and_caches.py
+cd -
 
 # pre-install the correct version of optimum for this release
 python -m pip install "optimum-graphcore>=0.5, <0.6"
 
 echo "Finished running setup.sh."
 # Run automated test if specified
-if [[ "$1" == "test" ]]; then
+if [[ "${1:-}" == "test" ]]; then
```

Comment:
Nit-picking to an incredible degree, I like the approach of using single (non-interpolating) quotes for fixed strings, to make it absolutely clear that we're referring to a literal value, e.g.

```bash
if [[ "${1:-}" == 'test' ]]; then
```